docs/db.md

```diff
@@ -1,4 +1,200 @@
 # Databases
 
-Explain how to integrate with Persistent.
-Explain also how to run a pool of database connections.
+In this section of the docs, to have a clearer understanding of how one would use `mu-haskell` to talk to a database, we are going to have a walk through the example of [`with-persistent`](https://github.com/higherkindness/mu-haskell/tree/master/examples/with-persistent).
+
+## First steps
+
+We are going to start with our source of truth: the proto file.
+
+```protobuf
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package withpersistent;
+
+message PersonRequest { int64 identifier = 1; }
+message Person { PersonRequest pid = 1; string name = 2; int32 age = 3; }
+
+service PersistentService {
+  rpc getPerson (PersonRequest) returns (Person);
+  rpc newPerson (Person) returns (PersonRequest);
+  rpc allPeople (google.protobuf.Empty) returns (stream Person);
+}
+```
+
+Maybe this example looks a bit contrived but bear with me, it covers a common use case when working with protobuf: that one of the messages has another message as its identifying key. Easier examples would not suffer from this and should be easier to implement.
```

Comment:
It's not very clear what "this" refers to in "easier examples would not suffer from this".